docker/service/src/integration-test/java/uk/nhs/adaptors/scr/uat/common/TestData.java

```diff
@@ -31,14 +31,15 @@ public static TestData build(List<Resource> resources) {
         var list = new ArrayList<>(resources);
         var testDataBuilder = TestData.builder();
 
-        var fhirResponse = list.stream()
+        list.stream()
             .filter(resource -> resource.getFilename() != null)
             .filter(resource -> resource.getFilename().endsWith(RESPONSE_FILE_ENDING))
             .findFirst()
-            .orElseThrow(() -> new IllegalStateException(String.format("Missing %s resource", RESPONSE_FILE_ENDING)));
+            .ifPresent(it -> {
+                testDataBuilder.fhirResponse(TestData.readFile(it));
+                list.remove(it);
```

Comment:
Will this work? Removing item while iterating?

Reply:
yes, works